modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java

```diff
@@ -225,8 +223,9 @@ public LazyMuleArtifactContext(MuleContext muleContext, ConfigResource[] artifac
     }
 
     initialize();
+    this.graph = generateFor(applicationModel);
```

Comment:
Put a comment at least: 
// Macro Expansion will generate new component models so we will create the graph at the end.